src/Node/Net.js

```diff
@@ -0,0 +1,403 @@
+"use strict";
+
+var net = require("net");
+
+exports.addressImpl = function (server) {
+  return function () {
+    return server.address();
+  };
+};
+
+exports.bufferSizeImpl = function (socket) {
+  return function () {
+    return socket.bufferSize;
+  };
+};
+
+exports.bytesRead = function (socket) {
+  return function () {
+    return socket.bytesRead;
+  };
+};
+
+exports.bytesWritten = function (socket) {
+  return function () {
+    return socket.bytesWritten;
+  };
+};
+
+exports.closeImpl = function (server) {
+  return function (callback) {
+    return function () {
+      server.close(function (err) {
+        callback(err)();
+      });
+    };
+  };
+};
+
+exports.connectImpl = function (socket) {
+  return function (options) {
+    return function (callback) {
+      return function () {
+        socket.connect(options, function () {
+          callback();
+        });
+      };
+    };
+  };
+};
+
+exports.connecting = function (socket) {
+  return function () {
+    return socket.connecting;
+  };
+};
+
+exports.createConnectionImpl = function (x) {
+  return function (callback) {
+    return function () {
+      return net.createConnection(x, function () {
```

Comment:
This could be written as `net.createConnection(x, callback)` right? That would avoid the extra function allocation each time too I think.

Reply:
I originally had it that way, but it was hard to know which callbacks took arguments and which didn't. Some of them weren't being invoked (the `Effect _` wasn't being run); the callback was passed two or three arguments in uncurried form where it was expecting those arguments curried and then to invoke the `Effect _` afterward.

Should this FFI be written with `Effect.Uncurried` instead? That seems like it would solve that problem and also remove the allocation. I hadn't seen `Effect.Uncurried` used in other FFI modules in this organization, so I wasn't sure if I should do it here.

Reply:
Yes, I think that would be better. The main reason we aren't using `Effect.Uncurried` elsewhere in node.js api bindings is that many of the bindings predate `Effect.Uncurried` I think.